### DISCUSSION TITLE
doc,docx,xls,xlsx,ppt,pptx无法预览的问题，但是pdf可以正常预览

### DISCUSSION THREAD

**Discussion by Andy81135** [AlistGo/alist#7179] [General]
在前面的提问中，发现修改对应Iframe 预览
比如你的链接是https://xxx/a.xlsx，iframe无论设置成src=$durl还是url=$e_url都是可以预览的，两个的区别是将url中的特殊符号转义
但是，如果你的链接是带有签名的，比如是在oss的私有空间存放，都会有特殊字符，如果使用durl，浏览器会认为&后是其他的参数，所以会浏览失败。如果用的是e_url，特殊字符转义后的链接，重新打开一个页面，是打不开的
这个问题，如果放在私有存储空间都会遇到，大家有什么解决方案么

**Comment by Andy81135** [comment]
pdf是可以正常预览的，因为pdf用的是另外一个组件

**Comment by Simonzhang8** [comment]
那$e_durl呢

**Reply by Andy81135** [reply]
e_durl的话，会将特殊字符如？、&转义，转义后的链接，放到浏览器里，其实是无法访问的

**Comment by Andy81135** [comment]
这个问题好像只会在本地存储的时候出现
